### PR TITLE
ISPN-8897 Indexes from different caches are mixed-up in server mode

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-8897

### CODE REVIEW

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProgrammaticSearchMappingProviderImpl.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
    public void defineMappings(Cache cache, SearchMapping searchMapping) {
       searchMapping.entity(ProtobufValueWrapper.class)
             .indexed()
+            .indexName(cache.getName() + "--" + ProtobufValueWrapperAnalyzerDiscriminator.class.getName())
```
Comment (gustavonalle) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProgrammaticSearchMappingProviderImpl.java]: Why not simply ``` .indexName(cache.getName())```? 
The index name is stored in each of entries of the lucene caches, so I'm a bit worried about large strings

Comment (anistor) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProgrammaticSearchMappingProviderImpl.java]: I just wanted to add some prefix or suffix to help distinguish it from possible other indexes that are user defined. A cache name usually comes from the user's domain so it is a potential conflict source. A shorter ```_protobuf``` suffix could work too.

**Comment by anistor** [comment]
Updated.

**Comment by anistor** [comment]
retest this please

**Comment by anistor** [comment]
retest this please

**Comment by anistor** [comment]
I've added the test from here https://github.com/infinispan/infinispan/pull/5809 to see what CI thinks about this combo.

**Comment by gustavonalle** [comment]
Merged, thanks @anistor !

**Pull request merged**